src/Generator/Passes/CheckAmbiguousFunctions.cs

```diff
@@ -172,6 +173,22 @@ private static bool CheckConstnessForAmbiguity(Function function, Function overl
             return false;
         }
 
+        private bool CheckConversionAmbiguity(Function function, Function overload)
+        {
+            var method1 = function as Method;
+            var method2 = overload as Method;
+            if (method1 == null || method2 == null)
+                return false;
+
+            if (method1.ReturnType.Type.ToString() == method2.ReturnType.Type.ToString())
```

Comment:
Please use `Type.Equals` instead.

Reply:
@ddobrev I tried that and it only works for `BuiltinType` types.

Reply:
If so, we have a bug. What did you try: `ReturnType.Equals` or `ReturnType.Type.Equals`?

Reply:
`ReturnType.Type` because `ReturnType` includes the `const` qualifier which I didn't care about.